### ISSUE TITLE
qwen2.5-7B GRPO训练时卡住，未显示任何报错

### ISSUE THREAD

**Issue by LiziLiziok** [modelscope/ms-swift#4603]
使用Docker: modelscope-registry.cn-hangzhou.cr.aliyuncs.com/modelscope-repo/modelscope:ubuntu22.04-cuda12.4.0-py311-torch2.6.0-vllm0.8.5.post1-modelscope1.26.0-swift3.4.1.post1

训练数据2000条

命令：
`
export NCCL_SOCKET_IFNAME=eth0 
export NCCL_IB_DISABLE=1 
export NCCL_DEBUG=INFO 
export NCCL_DEBUG_SUBSYS=COLL,GRAPH,INIT,ENV 
CUDA_VISIBLE_DEVICES=0 \
swift rollout \
  --model model_use/pre_train_model/Qwen2.5-7B-Instruct \
  --gpu_memory_utilization 0.7 \
  --tensor_parallel_size 1 \
  --data_parallel_size 1

`
`
export LOG_LEVEL=DEBUG 
export NCCL_DEBUG=INFO
export NCCL_SOCKET_IFNAME=lo
export WANDB_INIT_TIMEOUT=300
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6,7\
NPROC_PER_NODE=7 \
swift rlhf \
    --rlhf_type grpo \
    --model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --external_plugins /workspace/ouyang/model_test/reward/plugin.py \
    --reward_funcs random_reward \
    --reward_model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --reward_model_plugin my_rmplugin \
    --reward_weights 0.1 1 \
    --use_vllm true \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8001 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset model_test/data/GRPO/train_0613.jsonl \
    --max_length 12000 \
    --max_completion_length 20000 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 2e-6 \
    --gradient_accumulation_steps 4 \
    --eval_steps 18 \
    --save_steps 18 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --output_dir ./output/GRPO_COUNTDOWN \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 1 \
    --num_generations 2 \
    --temperature 1.0 \
    --system model_test/template/system.txt \
    --split_dataset_ratio 0.1 \
    --deepspeed zero3 \
    --log_completions true \
    --beta 0.001 \
    --num_iterations 1

`

这是卡住界面：

![Image](https://github.com/user-attachments/assets/86258ca9-5d8d-4c93-a0cc-d396aadf0e61)

**Comment by hjh0119** [comment]
can you retry with swift 3.5

**Comment by LiziLiziok** [comment]
> can you retry with swift 3.5

我尝试了：modelscope-registry.cn-beijing.cr.aliyuncs.com/modelscope-repo/modelscope:ubuntu22.04-cuda12.4.0-py311-torch2.6.0-vllm0.8.5.post1-modelscope1.27.0-swift3.5.1。
还是同样的问题，卡住，但是不报错。

我使用Colocate(Internal) Mode，也是卡住：
这是命令：
`
export LOG_LEVEL=DEBUG 
export NCCL_DEBUG=INFO
export NCCL_SOCKET_IFNAME=lo
export WANDB_INIT_TIMEOUT=300
export CUDA_LAUNCH_BLOCKING=1
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7\
NPROC_PER_NODE=8 \
swift rlhf \
    --rlhf_type grpo \
    --model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --external_plugins model_test/reward/plugin.py \
    --reward_funcs random_reward \
    --reward_model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --reward_model_plugin my_rmplugin \
    --reward_weights 0.1 1 \
    --use_vllm true \
    --vllm_mode colocate \
    --sleep_level 1\
    --offload_optimizer true \
    --offload_model true \
    --vllm_tensor_parallel_size 4 \
    --gc_collect_after_offload true \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset model_test/data/GRPO/train_0613.jsonl \
    --max_length 12000 \
    --max_completion_length 20000 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 2e-6 \
    --gradient_accumulation_steps 8 \
    --vllm_gpu_memory_utilization 0.5 \
    --eval_steps 10 \
    --save_steps 10 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --output_dir ./output/GRPO_COUNTDOWN \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 1 \
    --num_generations 2 \
    --temperature 1.0 \
    --system model_test/template/system.txt \
    --split_dataset_ratio 0.1 \
    --deepspeed zero3 \
    --log_completions true \
    --beta 0.001 \
    --num_iterations 1

`
这是卡住的界面：

![Image](https://github.com/user-attachments/assets/f55e510c-5a09-4475-9364-9a9676f40696)

![Image](https://github.com/user-attachments/assets/932cec5a-14e5-4c1c-830d-22ca0d23cb3e)


使用vllm_mode server：
这是命令：
`
export NCCL_SOCKET_IFNAME=eth0 
export NCCL_IB_DISABLE=1 
export NCCL_DEBUG=INFO 
export NCCL_DEBUG_SUBSYS=COLL,GRAPH,INIT,ENV 
CUDA_VISIBLE_DEVICES=0 \
swift rollout \
  --model /model_use/pre_train_model/Qwen2.5-7B-Instruct \
  --gpu_memory_utilization 0.7 \
  --tensor_parallel_size 1 \
  --data_parallel_size 1
`

`
export LOG_LEVEL=DEBUG 
export NCCL_DEBUG=INFO
export NCCL_SOCKET_IFNAME=lo
export WANDB_INIT_TIMEOUT=300
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6,7\
NPROC_PER_NODE=7 \
swift rlhf \
    --rlhf_type grpo \
    --model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --external_plugins ouyang/model_test/reward/plugin.py \
    --reward_funcs random_reward \
    --reward_model model_use/pre_train_model/Qwen2.5-7B-Instruct \
    --reward_model_plugin my_rmplugin \
    --reward_weights 0.1 1 \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset /workspace/ouyang/model_test/data/GRPO/train_0613.jsonl \
    --max_length 12000 \
    --max_completion_length 20000 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 2e-6 \
    --gradient_accumulation_steps 4 \
    --eval_steps 18 \
    --save_steps 18 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --output_dir ./output/GRPO_COUNTDOWN \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 1 \
    --num_generations 2 \
    --temperature 1.0 \
    --system model_test/template/system.txt \
    --split_dataset_ratio 0.1 \
    --deepspeed zero3 \
    --log_completions true \
    --beta 0.001 \
    --num_iterations 1

`
这是卡住的界面

![Image](https://github.com/user-attachments/assets/e0662e55-8a24-4937-b163-981402e94794)

![Image](https://github.com/user-attachments/assets/3fe0c936-51b3-4739-a79f-fae4b9fa6d59)

![Image](https://github.com/user-attachments/assets/db6065b4-9ecc-455c-be15-f8e32f89b94e)

**Comment by LiziLiziok** [comment]
将奖励模型去掉，是剩下奖励函数，恢复正常了。但是我需要使用到奖励模型，该怎么办

**Comment by hjh0119** [comment]
What is the implementation of my_rmplugin?

**Comment by LiziLiziok** [comment]
> What is the implementation of my_rmplugin?
我使用的是Qwen2.5-7B-Instruct
代码请看附件

[plugin.pdf](https://github.com/user-attachments/files/20794192/plugin.pdf)

**Comment by hjh0119** [comment]
I cannot reproduce the issue because there is garbled text in the PDF. Could you check if the genrm plugin within the framework is running properly? The implementation seems quite similar to me.

**Comment by Andre-110** [comment]
遇到了同样的问题，运行到一定程度就卡住了

<img width="1432" height="577" alt="Image" src="https://github.com/user-attachments/assets/cbeacc18-f0c7-4ec0-a223-4459eff9f81c" />

**Comment by puzhiyuan** [comment]
把zero3换成zero2试试，我用zero3也是会出现这样的情况

**Comment by robotsp** [comment]
我即使不用zero也会卡住 @puzhiyuan @Andre-110

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
